modules/csg/csg_shape.cpp

```diff
@@ -567,13 +604,17 @@ void CSGShape3D::_notification(int p_what) {
 				PhysicsServer3D::get_singleton()->free(root_collision_instance);
 				root_collision_instance = RID();
 				root_collision_shape.unref();
+				if (_is_debug_collision_shape_visible()) {
+					_clear_debug_collision_shape();
+				}
```

Comment:
I think toggling debug collision off (by calling `SceneTree::set_debug_collisions_hint(false)`) can lead to a leak, `root_collision_debug_instance` could end up not being cleared in such case.
```suggestion
				_clear_debug_collision_shape();
```